staging/src/k8s.io/apiserver/pkg/admission/plugin/cel/interface.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cel
+
+import (
+	"github.com/google/cel-go/common/types/ref"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/admission"
+)
+
+type PolicyDecisionKind string
+
+const (
+	Admit PolicyDecisionKind = "Admit"
+	Deny  PolicyDecisionKind = "Deny"
+)
+
+type FailurePolicy string
+
+const (
+	Fail   FailurePolicy = "Fail"
+	Ignore FailurePolicy = "Ignore"
+)
+
+type PolicyDecision struct {
+	PolicyDecisionKind `json:"kind"`
+	Message            interface{} `json:"message"`
+}
+
+type EvaluatorFunc func(a admission.Attributes, params *unstructured.Unstructured) []PolicyDecision
+
+// Object used by admission controller to convert runtime values into CEL-typed
+// values.
+type CELObjectConverter interface {
+	// Given a GVR, look up the openapi or JSONSchemaProps, structural schema, etc.
+	// and compile it into something that can be used to turn objects into CEL
+	// values
+	DeclForResource(gvr schema.GroupVersionResource) interface{}
+
+	// Object may be a typed native object or an unstructured object
+	ValueForObject(value interface{}, decls interface{}) *ref.Val
```

Comment:
For the value and decl types, do we have a better choice than `any`?

Reply:
decls has been updated. value may be either unstructured or a native type so I'm not sure it can be better than `any`

Reply:
Can be a `runtime.Object` I suppose?

Reply:
Yes, please use `runtime.Object`, that avoids the failure mode where someone passes a `Pod` instead of a `*Pod`...

Reply:
it's now been changed to `runtime.Object`